src/main/scala/ciris/aws/ssm/Region.scala

```diff
@@ -0,0 +1,73 @@
+package ciris.aws.ssm
+
+import cats.implicits._
+import ciris.ConfigDecoder
+import com.amazonaws.regions.Regions
+
+sealed abstract class Region(val asJava: Regions) {
+  final def name: String =
+    asJava.getName()
+
+  final def description: String =
+    asJava.getDescription()
+}
+
+object Region {
```

Comment:
why do we have this wrapper rather than just having an instance of `ConfigDecoder[String, Regions]` instead? as AWS has new regions all the time, we will need to keep this up to date. Also, in the client code which uses this lib, they might need to convert between Scala and Java

Reply:
> why do we have this wrapper rather than just having an instance of `ConfigDecoder[String, Regions]` instead?

The reason is in the pull request description:

> Add custom `Region` representation to enable decoding without importing any implicits.

Essentially, you would have to import that implicit instance into scope, which isn't great from a usability perspective. 

> as AWS has new regions all the time, we will need to keep this up to date.

We do indeed have to keep the list up-to-date, but we also benefit from exhaustiveness checking in the pattern match. Are you saying Amazon add new regions in patch versions? That sounds bad, because it's a binary incompatible change to do so.

> Also, in the client code which uses this lib, they might need to convert between Scala and Java

Not sure exactly what you're referring to here, except I agree it can be a slight nuisance, but you can still do total conversions.

Reply:
thanks for the well written explanation. Makes sense to me